ios/RNCAsyncStorage.m

```diff
@@ -156,6 +162,52 @@ static dispatch_queue_t RCTGetMethodQueue()
   return error ? RCTMakeError(@"Failed to delete storage directory.", error, nil) : nil;
 }
 
+
+NSString *OldStorageDirectory = @"RNCAsyncLocalStorage_V1";
+/**
+ * Creates an NSException used during Storage Directory Migration.
+ */
+static NSException *RCTStorageDirectionMigrationException(NSString *reason, NSError *error) {
+  NSMutableString *reasonString = [[NSMutableString alloc] initWithString:reason];
+  [reasonString appendString:@" - "];
+  [reasonString appendString:[error description]];
+  return [[NSException alloc] initWithName:@"RCTStorageDirectoryMigrationFailure" reason:reasonString userInfo:nil];
+}
+
+/**
+ * This check is added to make sure that anyone coming from pre-1.2.2 does not lose cached data.
+ * Data is migrated from the "RNCAsyncLocalStorage_V1" directory to the "RCTAsyncLocalStorage_V1" directory.
+ */
+static void RCTStorageDirectoryMigrationCheck() {
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    NSError *error;
+    BOOL isDir;
+    // If the old directory exists, it means we need to migrate data to the new directory
+    if ([[NSFileManager defaultManager] fileExistsAtPath:RCTCreateStorageDirectoryPath(OldStorageDirectory) isDirectory:&isDir] && isDir) {
+      // Check if the new storage directory location already exists
+      BOOL newStorageDirectoryExists = [[NSFileManager defaultManager] fileExistsAtPath:RCTGetStorageDirectory() isDirectory:&isDir];
+      if (newStorageDirectoryExists) {
+        // If the new storage directory location already exists, remove existing directory
+        newStorageDirectoryExists = !([[NSFileManager defaultManager] removeItemAtPath:RCTGetStorageDirectory() error:&error]);
+        if (newStorageDirectoryExists) {
+          @throw RCTStorageDirectionMigrationException(@"Failed to clear pre-existing storage directory", error);
+        }
+      }
+      if (!newStorageDirectoryExists) {
+        // If new storage direction doesn't exist, copy old storage directory to new location
+        if (![[NSFileManager defaultManager] copyItemAtPath:RCTCreateStorageDirectoryPath(OldStorageDirectory) toPath:RCTGetStorageDirectory() error:&error]) {
+          @throw RCTStorageDirectionMigrationException(@"Failed to copy old storage directory to new storage directory", error);
```

Comment:
We probably shouldn't take down the whole app if we fail to migrate. I think in the rare cases where IO might fail, maybe it's probably fine to just say that we "lost" the data?

Reply:
Yeah I am not a big fan of crashing the app either, but I'm not a fan of leaving errors unused either. I guess in this case it would be just fine to ignore the errors? Cause I don't think adding `NSLog`s will be of much use either.

Reply:
I think there's also a real risk of going into a crash loop. I think we should ignore it, and use [RCTLogWarn or RCTLogError](https://github.com/facebook/react-native/blob/629708bedae65a30e39d234da6b04d6fa101a779/React/Base/RCTLog.h) to log errors. What do you think, @Krizzu?

Reply:
Ah cool, didn't know these existed. Errors would also cause app crashes so I guess we could go with the Warnings to be safe. Right now I've replaced everything with NSLogs for now, so let me know what you guys think is best.

Reply:
I think I prefer `RCTLogWarn` because they will at least be visible during debug as a yellow box, in addition to being logged to the console.